File: backend/NoPressure.DAL/Entities/Team.cs
using NoPressure.Common.Enums;

namespace NoPressure.DAL.Entities
{
    public class Team
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime Date { get; set; }
        public string UniqId { get; set; }
        public EntityState State { get; set; }
        public TeamPrivacyState PrivacyState { get; set; }
        public string Color { get; set; }

        public int AuthorId { get; set; }
        public User Author { get; set; }
        public List<User> Users { get; set; }
        public List<Tag> Tags { get; set; }
        public List<TeamSettings> Settings { get; set; }
    }
}

File: frontend/src/app/models/team/team.ts
import { TeamAccess } from "../enums/TeamAccess";
import { TeamRole } from "../enums/TeamRole";
import { TeamTag } from "../tag/team-tag";
import { UserDTO } from "../user/user-dto";

export interface Team {
    id: number;
    name: string;
    date: Date;
    uniqId: string;
    color: string;
    addingUsers: TeamAccess;
    role: TeamRole;
    authorId: number;
    teamRequestId: number;
    users?: UserDTO[];
    tags?: TeamTag[];
}   